### PR TITLE
feat(docs): convert toc to .ts

### PR DESCRIPTION
**Pull request by deer**
To make conflicts easier to deal with, we should have a blank line at the end to essentially have a trailing comma. Now we can add a new entry _before_ the blank row without touching any other rows. Borrowed the idea from the showcase.

### CODE REVIEW

**Comment by deer** [comment]
or should this be `feat(www)`? 🤔

**Comment by lino-levan** [comment]
What worked for [denobyexample](https://github.com/denoland/denobyexample/blob/main/toc.ts) is just making the file a typescript file and exporting an object. JS allows trailing commas. I feel like that would be a cleaner solution.

**Comment by deer** [comment]
> What worked for [denobyexample](https://github.com/denoland/denobyexample/blob/main/toc.ts) is just making the file a typescript file and exporting an object. JS allows trailing commas. I feel like that would be a cleaner solution.

Good idea, that does sound better. Will update when I get a chance.

**Comment by marvinhagemeister** [comment]
+1 on turning it into `.ts` file. Makes typing easier too.

File: docs/toc.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-{
+import { RawTableOfContents } from "../www/data/docs.ts";
```
Comment (marvinhagemeister) [inline docs/toc.ts]: Nit: Let's move these types into this file to avoid circular imports.

**Review by marvinhagemeister — APPROVED** [review]
Great work! I like that this makes working with the TOC a little simpler 👍

**Pull request merged**